src/Shipping/ShippingController.php

```diff
@@ -60,6 +61,30 @@ function() {
 		add_filter( 'pre_update_option_pickup_location_pickup_locations', array( $this, 'flush_cache' ) );
 	}
 
+	/**
+	 * Gets a list of payment method ids that support the 'local_pickup' feature.
+	 *
+	 * @return string[] List of payment method ids that support the 'local_pickup' feature.
+	 */
+	public function get_local_pickup_method_ids() {
+		$methods_supporting_local_pickup = array_unique(
```

Comment:
I had to go back to doing it this way as `WC()->shipping()->get_shipping_methods()` returns like this:

<img width="563" alt="image" src="https://user-images.githubusercontent.com/5656702/216346852-1df5ff2b-66d3-464c-9714-11e5d3eeccb7.png">

For some reason it can include the same method multiple times, so stripping out duplicates and only outputting the method ID seems most sensible.

Is there a cleaner way of doing this?

Reply:
How about this:

```php
		$methods_supporting_local_pickup = array_reduce(
			WC()->shipping()->get_shipping_methods(),
			function( $methods, $method ) {
				if ( $method->supports( 'local_pickup' ) ) {
					$methods[] = $method->id;
				}
				return $methods;
			},
			array()
		);
```

Reply:
Nice, that looks better, but I still need to `array_unique` it, given `WC()->shipping()->get_shipping_methods()` can return the same one multiple times. I'll implement.